src/KonnektingDevice.cpp

```diff
@@ -514,7 +514,7 @@ void KonnektingDevice::handleMsgWriteProgrammingMode(byte msg[]) {
     }
 }
 
-void KonnektingDevice::handleMsgReadProgrammingMode(byte msg[]) {
+void KonnektingDevice::handleMsgReadProgrammingMode(byte []) {
```

Comment:
What's behind this argument name removal? Sure, the "msg" parameter is here not really used, but what is now "better"? Did the compiler complain about that unused variable? Never realized if so...

Reply:
When you switch on "All" warnings in the IDE, you will get a few warnings like
warning: unused parameter 'msg' [-Wunused-parameter]

I'd like to get rid of those "false positives" since they hide the few important warnings which point you to potential bugs during development. 
To avoid the unused parameter warnings, you can remove all parameter names from the function definitions (which are typically in the c/cpp file, or the inline functions in the headers). For the declaration in the headers it is good to keep them.

Reply:
Ah, I see. I never switched the default setting for the compiler warnings. 

But, on the one hand your (or we) try to make the code more clearly readable for new users (like the additional (...) on the inline-if), and then we create this "confusing"function signature by removing the parameter name. 

I think we should at least add a comment like this, to make it also for beginners more readable:

```
// to suppress compiler warning about unused parameter, the parameter name "msg" has been removed
```

But I would like to see a shorter comment here :-(
One the other side: Do you have experience with doxygen? How does the missing parameter name now work together with doxygen like this (picked a random function):

https://github.com/KONNEKTING/KonnektingDeviceLibrary/blob/develop_beta4a/src/KonnektingDevice.cpp#L573

I'm new to doxygen (trying to introduce this with beta4a onwards). My expectation would be, that doxygen complains, like javadoc would complain if the documentation is not exactly matching to the functions signature.

Reply:
Doxygen is a good choice, +1! This will help to keep the documentation in sync with the code, some stuff from the wiki could be transferred as doxyfiles.
Leaving out variable names is typically no problem since you are putting the doxygen tags at the declaration and leave out the names in the definition. Without using some tricks, for inlines it will probably be an undocumented but anyway unused parameter, or a compiler warning.